lib/registration.pm

```diff
@@ -256,6 +256,7 @@ sub fill_in_registration_data {
     # Process modules on sle 15
     if (is_sle '15+') {
         my $modules_needle = "modules-preselected-" . get_required_var('SLE_PRODUCT');
+        assert_screen($modules_needle);
```

Comment:
Hmm, I don't get this change =(. It will break some tests, as we need to get full list displayed before. So what is the problem with assertion later?

Reply:
The assertion tries to assert basesystem modules, server module and also that desktop is not selected, which seems to be unnecessary as the code later is doing exactly that. So my understanding is that the only intention that could be in that assertion could be to match before clicking in the checkbox (or even delete the assertion could be a good idea?).

Reply:
That's the thing, if not the park with unselected desktop, it would work, but we do this assertion when full list of modules is shown. In SLE 15 I know that behavior was changed to show pre-selected modules even if they are in beta phase. But this change will require thorough testing for existing scenarios including SLED, SLES4SAP, maybe HPC to see if needles are matching.

Reply:
I would like to avoid that heavy testing to be honest. I don't find useful that assertion for sle12sp4 and sle15sp1, what about `if is_sle('15-SP1+') || is_sle('12-sp4+')` and leave the heavy testing for the interested parties, as for us that code does not provide value (assert basesystem and server should not be needed, the important thing is the module that are added, but in case it would be needed? in the same way we could hack that with that previous condition and not affect other products). WDYT?

Reply:
right! :) ...while I am thinking better solution, I also saw that this worked and should not break any other tests: `send_key_until_needlematch $modules_needle, "down", 40;` VR: [sle-15-SP1-sles+sdk+proxy_SCC_via_YaST](http://dhcp42.suse.cz/tests/492#step/addon_products_via_SCC_yast2/29)